crates/teloxide/src/utils/render/mod.rs

```diff
@@ -0,0 +1,257 @@
+//! Util for rendering texts and message entities to html and markdown string
+
+use teloxide_core::types::{MessageEntity, MessageEntityKind as MEK};
+
+use tag::*;
+
+pub use helper::RenderMessageTextHelper;
+
+mod helper;
+mod html;
+mod markdown;
+mod tag;
+
+/// The [`Render`] struct is responsible for parsing the text and entities to
+/// produce the final formatted output.
+#[derive(Clone, Eq, PartialEq)]
+pub struct Render<'a> {
+    text: &'a str,
+    tags: Vec<Tag<'a>>,
+}
+
+impl<'a> Render<'a> {
+    /// Creates a new `Render` instance with the given text and entities.
+    ///
+    /// The `Render` is responsible for parsing the text and entities to
+    /// produce the final formatted output. This constructor sets up the
+    /// initial state needed for the parsing process.
+    ///
+    /// # Arguments
+    ///
+    /// - `text`: The input text to be parsed.
+    /// - `entities`: The message entities (formatting, links, etc.) to be
+    ///   applied to the text.
+    ///
+    /// # Returns
+    ///
+    /// A new [`Render`] instance.
```

Comment:
Well, this is already evident from the signature, so I wouldn't include it in the documentation.